hrms/hr/doctype/shift_assignment/shift_assignment.py

```diff
@@ -35,6 +40,29 @@ def validate_overlapping_shifts(self):
 			if overlapping_timings:
 				self.throw_overlap_error(overlapping_dates[0])
 
+	def validate_same_date_multiple_shifts(self):
```

Comment:
Can move this check after line no 37 instead of running overlapping dates query twice. pass `overlapping_dates` to this function.